Can't use crate as library

I wanted to use this project as a library in a project of mine, but because there's no `lib.rs`, it seems that I can't actually use this crate without vendoring the source code in my project.

```
$ cargo build
   Compiling my-project v0.1.0 (/home/nick/Documents/code/my-project)
error[E0432]: unresolved import `speedtest_rs`
 --> src/main.rs:1:5
  |
1 | use speedtest_rs::*;
  |     ^^^^^^^^^^^^ use of undeclared type or module `speedtest_rs`
```

Would you consider adding a `lib.rs`? I saw in another issue you don't want to commit to a specific API at the moment, but I'd love to be able to use this code as a dependency, rather than just vendored source.

--- a/README.md
+++ b/README.md
@@ -11,7 +11,7 @@ Status: This is usable for lower-end residential connections using ["HTTP Legacy
 
 This tool currently only supports [HTTP Legacy Fallback][http_legacy_fallback] for testing.
 
-High bandwidth connections higher than ~200Mbps may return incorrect results! 
+High bandwidth connections higher than ~200Mbps may return incorrect results!
 
 The testing operations are different from socket versions of tools connecting to speedtest.net infrastructure. In the many FOSS Go versions, tests are done to find an amount of data that can run for a default of 3 seconds over some TCP connection. In particular, `speedtest-cli` and `speedtest-rs` tests with what Ookla calls the ["HTTP Legacy Fallback"][http_legacy_fallback] for hosts that cannot establish a direct TCP connection.
 
@@ -22,7 +22,7 @@ The testing operations are different from socket versions of tools connecting to
 
 https://www.speedtest.net/apps/cli
 
-Please look here. Unfortunately, it is not FOSS. Still, it is supported by them and can be used for non-commercial purposes. 
+Please look here. Unfortunately, it is not FOSS. Still, it is supported by them and can be used for non-commercial purposes.
 
 ## Purpose
 
@@ -32,11 +32,15 @@ The [HTTP Legacy Fallback][http_legacy_fallback] is currently based on the popul
 
 https://github.com/sivel/speedtest-cli @ 2.1.2
 
-There are also other speedtest.net using tools using different approaches to be stolen from in the future. For example: 
+There are also other speedtest.net using tools using different approaches to be stolen from in the future. For example:
 
 https://github.com/traetox/speedtest
 
 This example seems different as it appears to just use TCP connections and some protocol. It's probably more suitable to high-speed connections. TODO: Add a default TCP-mode.
+
+## Use as a Library
+
+The API is very much not stable. Use at your own risk. Semver adherence definitely not guaranteed. Please lock to exact versions if you must.
 
 ## License
 

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -1,0 +1,8 @@
+// This crate really isn't meant to be stable.
+
+pub mod distance;
+pub mod error;
+pub mod speedtest;
+pub mod speedtest_config;
+pub mod speedtest_csv;
+pub mod speedtest_servers_config;

--- a/src/main.rs
+++ b/src/main.rs
@@ -1,4 +1,8 @@
+mod distance;
+mod error;
+mod speedtest;
 mod speedtest_config;
+mod speedtest_csv;
 mod speedtest_servers_config;
 
 use crate::speedtest_csv::SpeedTestCsvResult;
@@ -7,11 +11,6 @@ use clap::{crate_version, App, Arg};
 use log::info;
 use std::io::{self, Write};
 use url::Url;
-
-mod distance;
-mod error;
-mod speedtest;
-mod speedtest_csv;
 
 fn main() -> Result<(), error::Error> {
     env_logger::init();

